src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java

```diff
@@ -1,35 +1,40 @@
 package suparnaNJune20.Assignment_31;
+
 import java.util.InputMismatchException;
 import java.util.Scanner;
+
 public class InputNumberExceptionHandling {
-	static int sum = 0, cnt = 0;
-	public void sumOfNumber(int inputCnt)
-	{
-		
-		
+	int sum = 0, cnt = 0, inputNo = 0;;
+
+	public void sumOfNumber(int inputCnt) {
 		Scanner scanner = new Scanner(System.in);
-		for ( cnt = 1; cnt <= inputCnt; cnt++) {
-			
+		StringBuffer inputStr = new StringBuffer("");
+		for (cnt = 1; cnt <= inputCnt; cnt++) {
+
 			try {
-				System.out.println("Enter "+ cnt +" input Number ");
-				sum += scanner.nextInt();
-				
+				System.out.println("Enter " + cnt + " input Number ");
+				inputNo = scanner.nextInt();
+				sum += inputNo;
+				inputStr = inputStr.append(inputNo);
+				inputStr = inputStr.append(" , ");
+
 			} catch (InputMismatchException inputMismatchException) {
 				System.out.println("Enter Number input not a string  ");
 				cnt--;
-				scanner.next();
-				
+				inputStr = inputStr.append(scanner.next());
```

Comment:
Here, if we provide invalid input (eg: abc) then inputStr still contains that invalid input. then it gets appended with the next input given.